plugins/kubernetes-backend/src/service/router.ts

```diff
@@ -17,19 +17,70 @@
 import express from 'express';
 import Router from 'express-promise-router';
 import { Logger } from 'winston';
+import { Config } from '@backstage/config';
+import {
+  ClusterLocatorMethod,
+  KubernetesClusterLocator,
+} from '../cluster-locator/types';
+import { MultiTenantConfigClusterLocator } from '../cluster-locator/MultiTenantConfigClusterLocator';
+import {
+  KubernetesClientBasedFetcher,
+  KubernetesFetcher,
+} from './KubernetesFetcher';
+import { KubernetesClientProvider } from './KubernetesClientProvider';
+import {
+  GetKubernetesObjectsByServiceIdHandler,
+  handleGetKubernetesObjectsByServiceId,
+} from './getKubernetesObjectsByServiceIdHandler';
 
 export interface RouterOptions {
   logger: Logger;
+  config: Config;
 }
 
-const makeRouter = (logger: Logger): express.Router => {
+const getClusterLocator = (config: Config): KubernetesClusterLocator => {
+  const clusterLocatorMethod = config.getString(
+    'kubernetes.clusterLocatorMethod',
+  ) as ClusterLocatorMethod;
+
+  switch (clusterLocatorMethod) {
+    case 'configMultiTenant':
+      return MultiTenantConfigClusterLocator.readConfig(
+        config.getConfigArray('kubernetes.clusters'),
+      );
+    case 'http':
+      throw new Error('not implemented');
+    default:
+      throw new Error(
+        `Unsupported kubernetes.clusterLocatorMethod "${clusterLocatorMethod}"`,
+      );
+  }
+};
+
+export const makeRouter = (
```

Comment:
Are you thinking that `makeRouter` is a lower-level option where you pass in your own implementations, while `createRouter` is higher-level and config-driven?

Reply:
Yeah, make router for unit tests, create router for e2e tests

Reply:
Both are exported from the plugin though, so they are public API that can be used in the backend package. I think it makes sense though, but we may want to establish a pattern around it. Nothing for this PR though :grin: